Uplift/Controllers/ActivitiesViewController.swift

```diff
@@ -0,0 +1,116 @@
+        headerView.layer.shadowOffset = CGSize(width: 0.0, height: 4.0)
+        headerView.layer.shadowOpacity = 0.4
+        headerView.layer.shadowRadius = 10.0
+        headerView.layer.shadowColor = UIColor.buttonShadow.cgColor
+        headerView.layer.masksToBounds = false
+        view.addSubview(headerView)
+
+        let activitiesFlowLayout = UICollectionViewFlowLayout()
+        activitiesFlowLayout.scrollDirection = .vertical
+        activitiesFlowLayout.minimumInteritemSpacing = 150
+        activitiesCollectionView = UICollectionView(frame: .zero, collectionViewLayout: activitiesFlowLayout)
+
+        activitiesCollectionView.register(ActivityCollectionViewCell.self, forCellWithReuseIdentifier: Identifiers.activitesReuseIdentifier)
+        activitiesCollectionView.delegate = self
+        activitiesCollectionView.dataSource = self
+        view.addSubview(activitiesCollectionView)
+
+        filterButton.setTitle("FILTER", for: .normal)
+        filterButton.titleLabel?.font = ._16MontserratBold
+        filterButton.setTitleColor(.black, for: .normal)
+        filterButton.layer.shadowColor = UIColor.buttonShadow.cgColor
+        filterButton.layer.shadowOffset = CGSize(width: 0, height: 2)
+        filterButton.layer.shadowRadius = 4
+        filterButton.backgroundColor = .white
+        filterButton.layer.cornerRadius = 20
+        filterButton.layer.shadowOpacity = 1
+        view.addSubview(filterButton)
+    }
+
+    private func setupConstraints() {
+        let bottomPadding = 100
+        let headerViewHeight = 120
+        let filterWidth = 100
+        let filterHeight = 41
+        let filterPadding = 15
+
+        headerView.snp.makeConstraints { make in
+            make.leading.trailing.top.equalToSuperview()
+            make.height.equalTo(headerViewHeight)
+        }
+
+        activitiesCollectionView.snp.makeConstraints { make in
+            make.leading.equalToSuperview()
+            make.trailing.equalToSuperview()
+            make.top.equalTo(headerView.snp.bottom)
+            make.bottom.equalToSuperview().inset(-bottomPadding)
+        }
+
+        filterButton.snp.makeConstraints { make in
+            make.top.equalTo(activitiesCollectionView.snp.bottom).offset(filterPadding)
+            make.width.equalTo(filterWidth)
+            make.height.equalTo(filterHeight)
+            make.centerX.equalToSuperview()
+        }
+    }
+
+}
+
+extension ActivitesViewController: UICollectionViewDelegateFlowLayout {
+
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
+        let padding: CGFloat = 10
+        return CGSize(width: (collectionView.bounds.width - padding * 2), height: 60)
+    }
+
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, minimumLineSpacingForSectionAt section: Int) -> CGFloat {
+        return 20
```

Comment:
You could filter out these magic numbers to a private `Constants` struct in an extension to this class.